core/primitives/src/version.rs

```diff
@@ -61,3 +61,28 @@ impl ProtocolVersionRange {
         self.lower <= version && self.upper.map_or(true, |upper| version < upper)
     }
 }
+
+/// New Protocol features should go here. Nightly features are guarded by `#[cfg(feature = "nightly_protocol")]`.
+pub enum ProtocolFeature {}
+
+/// Current latest stable version of the protocol.
+#[cfg(not(feature = "nightly_protocol"))]
+pub const PROTOCOL_VERSION: ProtocolVersion = 40;
+
+/// Current latest nightly version of the protocol.
+#[cfg(feature = "nightly_protocol")]
+pub const PROTOCOL_VERSION: ProtocolVersion = 40;
+
+#[cfg(not(feature = "nightly_protocol"))]
+lazy_static! {
+    static ref PROTOCOL_VERSION_FEATURES_MAPPING: HashMap<ProtocolVersion, Vec<ProtocolFeature>> = vec![
+        /* add mapping here */
+    ].into_iter().collect();
```

Comment:
Well, we seem to agree that we cannot disable features, which means that all the features are additive over protocol versions, and thus I think we should better have a mapping of ProtocolFeature -> Minimal ProtocolVersion for quick lookup otherwise every protocol version will have duplicated information about the set of supported features accumulated from the previous protocol versions.

Reply:
No we need the protocol version to feature lookup because when you use some features, the way to tell whether it should be enabled is by grabbing the current protocol version and see what features are enabled in this protocol version

Reply:
@bowenwang1996 Won't this be sufficient?

```rust
if current_protocol_version >= PROTOCOL_VERSION_BY_FEATURE[MyFeature]
```

Reply:
ah good point